packages/react-native-web/src/exports/Modal/ModalFocusTrap.js

```diff
@@ -0,0 +1,151 @@
+/**
+ * Copyright (c) Nicolas Gallagher.
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @flow
+ */
+
+import React, { useRef, useCallback, useEffect } from 'react';
+
+import { canUseDOM } from 'fbjs/lib/ExecutionEnvironment';
+
+import View from '../View';
+import createElement from '../createElement';
+import StyleSheet from '../StyleSheet';
+
+/**
+ * This Component is used to "wrap" the modal we're opening
+ * so that changing focus via tab will never leave the document.
+ *
+ * This allows us to properly trap the focus within a modal
+ * even if the modal is at the start or end of a document.
+ */
+
+const FocusBracket = () => {
+  return createElement(
+    'div',
+    {
+      style: styles.focusBracket,
+      accessibilityRole: 'none',
+      focusable: true,
+      tabIndex: 0
```

Comment:
You don't need this

Reply:
TLDR: Per testing, If `visibility` is set to `hidden` or `tabIndex` is unset it seems to break the focus trapping.
Without these being set they no longer are focusable after the elements within the modal.

The focus swaps from within the modal to outside the modal when you hit `tab` within the
modal.  Great!  Normally the focus trap catches & we move the focus back to the top.
This is done by listening to focus events on the document / user agent browser pane.

Except.. If the modal is the "last" focusable item on the page.  At that point the focus leaves
the user agent - to the browser chrome instead.  The focus event on the document never fires
for this because we aren't changing focus to another element within the document - we leave the document.

That's what the focus brackets are there to prevent - to always force an element leaving the modal to find another
element to focus on before leaving the document.

Not sure how to write a test to verify this, though.  Perhaps render the document with the focus trap enabled and search up and down from modal that there's at least one focusable element outside of the div?

Reply:
Happy to remove the `tabIndex` and set `visibility: hidden` so we can verify this in storybook

Reply:
Oh yeah I haven't added focusable to this library yet, it's `accessible`. And the focus bracket should be visually hidden using the `clip` technique and probably hidden from screen readers too. There probably needs to be a bit of hidden ui that is available to screen readers and can be used to close the modal

Reply:
> Oh yeah I haven't added focusable to this library yet, it's accessible.

I'll play with `accessible` a bit.

>  And the focus bracket should be visually hidden using the clip technique

Isn't it already hidden, though?  It has no content, no height, etc.  We can definitely clip it to hide it further, though, just want to make sure that's not overkill here.

> and probably hidden from screen readers too.

Sure, that makes sense. I'll add `aria-hidden` to the element.

> There probably needs to be a bit of hidden ui that is available to screen readers and can be used to close the modal

We do have the <kbd>Esc</kbd> key to close the dialog which might fulfill that purpose for screen readers.  The only other bit suggested per WAI-ARIA is ["It is strongly recommended that the tab sequence of all dialogs include a visible element with role button that closes the dialog, such as a close icon or cancel button."](https://www.w3.org/TR/wai-aria-practices-1.1/#keyboard-interaction-7) - but that refers to a visible elements which would affect the UI & design.

I don't believe the react-native implementation has much more than this either, but that was just through a cursory look [at the javascript bits](https://github.com/facebook/react-native/blob/master/Libraries/Modal/Modal.js) - but most handheld devices or TVs have a back button that might satisfy this.

Reply:
To get `aria-hidden` I used `importantForAccessibility='no-hide-descendants'`